test/unit/lodash.spec.js

```diff
@@ -0,0 +1,690 @@
+            });
+            it('should return true when predicates with fromIndex find the value', function () {
+                expect(_.includes.apply(_, [array, predicate1[0], predicate1[1]])).to.eql(true);
+                expect(_.includes.apply(_, [array, predicate2[0], predicate2[1]])).to.eql(true);
+                expect(_.includes.apply(_, [array, predicate3[0], predicate3[1]])).to.eql(true);
+            });
+        });
+    });
+
+    describe('map', function () {
+        describe('map function', function () {
+            var params = datasets.map();
+
+            it('should match types of passing elements', function () {
+                var element = typeof params[1];
+
+                expect(element).to.eql('string');
+                expect(params[1]).to.have.length.above(0);
+                expect(params.length).to.eql(2);
+            });
+            it('should map by one of the property name', function () {
+                var testFunction = function () {
+                    if (Math.random() < 0.2) {
+                        return {firstName: 'Ali', lastName: 'BOM', user: false};
+                    } else if (Math.random() < 0.4) {
+                        return {firstName: 'Kowal', lastName: 'Kot', user: true};
+                    } else if (Math.random() < 0.6) {
+                        return {firstName: 'Buro', lastName: 'Fisher', user: true};
+                    } else if (Math.random() < 0.8) {
+                        return {firstName: 'Barney', lastName: 'Sick', user: false};
+                    } else {
+                        return {firstName: 'Draco', lastName: 'Malio', user: true};
+                    }
+                };
+                var testArray = _.fill(new Array(10), 1).map(testFunction);
+                var params = datasets.map(testArray);
+                var array = params[0];
+                var result = _.map.apply(_, params);
+                expect(result).to.not.include(undefined);
+                expect(array.length).to.eql(10);
+                expect(_.map.apply(_, datasets.map(testArray))).to.eql(result);
+            });
+        });
+        describe('map2 function', function () {
+            var params = datasets.map2();
+            var spy = sinon.spy(params[1]);
+
+            it('should match types of passing elements', function () {
+                var element1 = params[0] instanceof Array;
+                var element2 = params[1] instanceof Function;
+
+                expect(element1).to.eql(true);
+                expect(element2).to.eql(true);
+                expect(params.length).to.eql(2);
+            });
+            it('should return false if array contains not only number', function () {
+                spy('as');
+                expect(spy).to.have.returned(false);
+                expect(spy).to.have.been.callCount(1);
+            });
+            it('should pass mock test', function () {
```

Comment:
You are basically makeing exercise harder without serious purpose of it I think. Task title should be more precise here and moreover I was once again able to hardcode all answers without using any square functions, but overally it is much better than before(challenging at least).